src/test/java/com/codurance/retropolis/applicationservices/ApplicationCardServiceTest.java

```diff
@@ -0,0 +1,120 @@
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+@ExtendWith(MockitoExtension.class)
+public class ApplicationCardServiceTest {
+
+  private final Boolean HAVE_VOTED = false;
+  private final String EMAIL = "john.doe@codurance.com";
+  private final User USER = new User(3L, "john.doe@codurance.com", "John Doe");
+  private final Long CARD_ID = 1L;
+  private final Long COLUMN_ID = 2L;
+  private final String TEXT = "Hello";
+  @Mock
+  UserService userService;
+  @Mock
+  CardService cardService;
+  @Mock
+  CardResponseObjectFactory cardResponseObjectFactory;
+  private ApplicationCardService applicationCardService;
+  private NewCardRequestObject newCardRequestObject;
+  private UpVoteRequestObject upvoteRequestObject;
+
+
+  @BeforeEach
+  void setUp() {
+    applicationCardService = new ApplicationCardService(userService, cardService,
+        cardResponseObjectFactory);
+
+    newCardRequestObject = new NewCardRequestObject(TEXT, COLUMN_ID, USER.email);
+  }
+
+  @Test
+  void creates_card_response_object() {
+    Card card = new Card(CARD_ID, TEXT, COLUMN_ID, USER.getId(), emptyList());
+
+    User author = new User(USER.getId(), USER.email, USER.username);
+    CardResponseObject cardResponseObject = new CardResponseObject(card.getText(), card.getId(),
+        card.getColumnId(), HAVE_VOTED, card.getVoters().size(), author.username);
+
+    when(userService.findByEmail(newCardRequestObject.getEmail())).thenReturn(author);
+    when(cardService.create(newCardRequestObject)).thenReturn(card);
+
+    when(userService.findById(author.getId())).thenReturn(author);
+    when(cardResponseObjectFactory.create(card, author.getId(), author.username))
+        .thenReturn(cardResponseObject);
+
+    CardResponseObject response = applicationCardService.create(newCardRequestObject);
+
+    assertEquals(cardResponseObject.getId(), response.getId());
+    assertEquals(cardResponseObject.getAuthor(), response.getAuthor());
+    assertEquals(cardResponseObject.getText(), response.getText());
+  }
+
+  @Test
+  void adds_upvote_to_card_response() {
+    upvoteRequestObject = new UpVoteRequestObject(EMAIL, true);
+    ResponseEntity<HttpStatus> response = applicationCardService
+        .addUpvote(CARD_ID, upvoteRequestObject);
+    assertEquals(HttpStatus.OK, response.getStatusCode());
+  }
+
+  @Test
+  void removes_upvote_from_card_response() {
+    upvoteRequestObject = new UpVoteRequestObject(EMAIL, false);
+    ResponseEntity<HttpStatus> response = applicationCardService
+        .removeUpvote(CARD_ID, upvoteRequestObject);
+    assertEquals(HttpStatus.OK, response.getStatusCode());
+  }
+
+  @Test
+  void deletes_card_with_id() {
+    ResponseEntity<HttpStatus> response = applicationCardService.delete(CARD_ID);
+    verify(cardService).delete(CARD_ID);
```

Comment:
Here you verify the flow of the function but on the above test `removes_upvote_from_card_response` you just test the response without verifying, is there any particular reason for that?

Reply:
Sorry. Can you clarify.

Reply:
`deletes_card_with_id` you use `verify` to test the flow and then you `assert`  on the response. on the test `removes_upvote_from_card_response` you only `assert` the response
this is inconsistent, is there any reason for that?
either only `assert` the result or use `verify` on the flow

Reply:
Thanks. Understood.